Core/Src/state_machine.c

```diff
@@ -2,23 +2,23 @@
 #include "can_handler.h"
 #include "fault.h"
 #include "nero.h"
-#include "pdu.h"
 #include "queues.h"
 #include "monitor.h"
 #include "serial_monitor.h"
 #include "nero.h"
-#include "pdu.h"
 #include "queues.h"
+#include "processing.h"
 #include <stdbool.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include "cerb_utils.h"
 
 #define STATE_TRANS_QUEUE_SIZE 4
+#define STATE_TRANSITION_FLAG  1U
 
 /* Internal State of Vehicle */
 static state_t cerberus_state;
 
-static pdu_t *pdu;
-
 extern IWDG_HandleTypeDef hiwdg;
 
 osTimerId fault_timer;
```

Comment:
where is this timer initialized?

Reply:
Nowhere idk how we didnt catch this till now I guess RTOS just ignored the null

Reply:
Then how did we ever unfault?

Reply:
Or what if we only unfaulted by a crash and watchdog just from this assert failure?

Reply:
I think everytime we faulted we crashed cuz of the stack overflow error. Do you know what the purpose of this timer is? The comment does not explain it well.

Reply:
I thought it was the timer to unfault (i.e.) clear faults and re-enter ready.

Reply:
This was init was removed in peytons recent state machine refactor so nothing to worry about really as long as it's brought back.

Reply:
Ok we will do away with it for now cuz you can re enter ready by just hitting the home button.

Reply:
So how do we clear faults now?